cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/CertificateRenewalTest.java

```diff
@@ -102,6 +109,18 @@ public class CertificateRenewalTest {
 
     private List<Secret> secrets = new ArrayList<>();
 
+    @BeforeAll
+    public static void before() {
+        vertx = Vertx.vertx();
```

Comment:
I think by putting this in BeforeAll rather than BeforeEach it effectively changes the scope of the vertx object. My understanding is that if you pass the vertx object as an argument to each test you will get a new instance for each test, whereas here the result would be one instance shared by all tests. Perhaps not a problem but worth making sure we are happy with that.

Reply:
Correct, I checked and it's constructing a new instance per test.

I don't think it's a problem, in that the tests are passing. It was changed from static to using the extension to inject [as part of a broad test tidy up PR](https://github.com/strimzi/strimzi-kafka-operator/pull/5204/files#diff-42e49bd8eda8159c300cf6bab463e00bdd7f8c5ad67c2e6643fde3b0fdddf16e). I couldn't see a discussion on it so at a guess @LiamClarkeNZ was removing some code that could be handled for us by the framework. Do you remember at all @LiamClarkeNZ?

Maybe we could come back around with another PR to pick a style (static vs injected-per-test) and potentially pull out a base test class.

Reply:
@robobario @katheris As this is running under the `VertxExtension`, and there are no test specific config changes being made in `beforeAll()`, manually instantiating a `Vertx()` isn't necessary. 

Just add a `Vertx` argument to test methods that need to interact with a Vertx instance, and the extension will inject it automatically.

Reply:
I rolled it back to use injection, we still need to keep an executor reference per test so the field is created in a BeforeEach and closed in an AfterEach